packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,384 @@
+    }
+
+    function output() public view returns (Output memory) {
+        return Output({
+            superchainProxyAdmin: superchainProxyAdmin,
+            superchainConfigImpl: superchainConfigImpl,
+            superchainConfigProxy: superchainConfigProxy,
+            protocolVersionsImpl: protocolVersionsImpl,
+            protocolVersionsProxy: protocolVersionsProxy
+        });
+    }
+
+    function checkOutput() public view {
+        // Assert that all addresses are non-zero and have code.
+        // We use LibString to avoid the need for adding cheatcodes to this contract.
+        address[] memory addresses = new address[](5);
+        addresses[0] = address(superchainProxyAdmin);
+        addresses[1] = address(superchainConfigImpl);
+        addresses[2] = address(superchainConfigProxy);
+        addresses[3] = address(protocolVersionsImpl);
+        addresses[4] = address(protocolVersionsProxy);
+
+        for (uint256 i = 0; i < addresses.length; i++) {
+            address who = addresses[i];
+            require(who != address(0), string.concat("check failed: zero address at index ", LibString.toString(i)));
+            require(
+                who.code.length > 0, string.concat("check failed: no code at ", LibString.toHexStringChecksummed(who))
+            );
+        }
+
+        // All addresses should be unique.
+        for (uint256 i = 0; i < addresses.length; i++) {
+            for (uint256 j = i + 1; j < addresses.length; j++) {
+                string memory err =
+                    string.concat("check failed: duplicates at ", LibString.toString(i), ",", LibString.toString(j));
+                require(addresses[i] != addresses[j], err);
+            }
+        }
+    }
+}
+
+// For all broadcasts in this script we explicitly specify the deployer as `msg.sender` because for
+// testing we deploy this script from a test contract. If we provide no argument, the foundry
+// default sender is be the broadcaster during test, but the broadcaster needs to be the deployer
+// since they are set to the initial proxy admin owner.
+contract DeploySuperchain is Script {
+    // -------- Core Deployment Methods --------
+
+    // This entrypoint is for end-users to deploy from an input file and write to an output file.
+    // In this usage, we don't need the input and output contract functionality, so we deploy them
+    // here and abstract that architectural detail away from the end user.
+    function run(string memory _infile) public {
+        // End-user without file IO, so etch the IO helper contracts.
+        DeploySuperchainInput dsi = DeploySuperchainInput(toIOAddress(msg.sender, "optimism.DeploySuperchainInput"));
+        DeploySuperchainOutput dso = DeploySuperchainOutput(toIOAddress(msg.sender, "optimism.DeploySuperchainOutput"));
+        vm.etch(address(dsi), type(DeploySuperchainInput).runtimeCode);
+        vm.etch(address(dso), type(DeploySuperchainOutput).runtimeCode);
```

Comment:
It may be worth putting this in a helper function, to de-dup with the other one.

And the dsi/dso addresses could maybe be derived from the self-address, if each unit-test already deploys a unique instance of `DeploySuperchain`.

Reply:
De-duped in 9cf7aec4a and added a more explicit getter for how the dsi/dso addresses are derived